libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt

```diff
@@ -772,6 +774,28 @@ class MapboxTripSessionTest {
         tripSession.stop()
     }
 
+    @Test
+    fun voiceInstructionsFollowRouteProgress() = coroutineRule.runBlockingTest {
+        val voiceInstructionsObserver: VoiceInstructionsObserver = mockk(relaxUnitFun = true)
+        val voiceInstructions: VoiceInstructions = mockk()
+        val routeProgressObserver: RouteProgressObserver = mockk(relaxUnitFun = true)
+        every { routeProgress.voiceInstructions } returns voiceInstructions
+
+        tripSession = buildTripSession()
+        tripSession.start(true)
+        tripSession.registerVoiceInstructionsObserver(voiceInstructionsObserver)
+        tripSession.registerRouteProgressObserver(routeProgressObserver)
+
+        navigatorObserverImplSlot.captured.onStatus(navigationStatusOrigin, navigationStatus)
+
+        every { routeProgress.voiceInstructions } returns null
+
+        navigatorObserverImplSlot.captured.onStatus(navigationStatusOrigin, navigationStatus)
+
+        verify(exactly = 1) { voiceInstructionsObserver.onNewVoiceInstructions(any()) }
+        verify(exactly = 2) { routeProgressObserver.onRouteProgressChanged(any()) }
+    }
```

Comment:
This is the test I don't trust. I leave it just for demonstration. Going to remove after feedback.

Reply:
what do you mean you don't trust it?

Reply:
I mean the test doesn't give me confidence that feature works as expected. I know the logic which maps `NavigationStatus` to `RouteProgress` is mocked. The only real part that test verifies is 
![image](https://user-images.githubusercontent.com/6190346/142989358-fcd126b4-49a5-446c-ab85-cf30c0bd7647.png)

Reply:
But maybe confidence isn't the main problem for me.

The main problem is I don't know how to work using TDD when I mock so many things. I want to state in the test the feature requirements "RouteProgress.voiceInstruction shouldn't be null if possible. When native navigator returns null in a status update, fallback to previous value but don't trigger the new voice instruction event". All I can do in this test is "When mapper returns null voice instruction trigger route progress event but don't trigger new voice instruction event".

Reply:
@LukasPaczos , the mapping is already tested in the bigger `voiceInstructionsFallbacksToPreviousValue` test. Do you think it's worth to test it one more time in the `NavigatorMapperTest` ?

Reply:
You wouldn't have this question if the mapping was mocked in TripSession tests :)